test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1664,4 +1658,97 @@ public void unblockSequencerRecoveryOnDeadlock() throws Exception {
         getManagementServer(SERVERS.PORT_0).getManagementAgent().triggerSequencerBootstrap(layout)
                 .get();
     }
+
+    /**
+     * Scenario to verify that we do not regress the token count when the layout switches primary
+     * sequencers.
+     * We assert that the failover sequencer should always receive a full bootstrap message rather
+     * than an empty bootstrap message (without streamTailsMap)
+     */
+    @Test
+    public void failoverSeqDoesNotRegressTokenValue() throws Exception {
+        getManagementTestLayout();
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+
+        // Append data.
+        IStreamView streamView = corfuRuntime.getStreamsView()
+                .get(CorfuRuntime.getStreamID("testStream"));
+        final byte[] payload = "test".getBytes();
+        final int num = 10;
+        // 0 - 9
+        for (int i = 0; i < num; i++) {
+            streamView.append(payload);
+        }
+
+        // Adding rule to partition out PORT_0.
+        addServerRule(SERVERS.PORT_0, new TestRule().always().drop());
+
+        // Append data 10 - 19
+        for (int i = 0; i < num; i++) {
+            streamView.append(payload);
+        }
+
+        // Remove partition and allow PORT_0 to be healed.
+        clearServerRules(SERVERS.PORT_0);
+
+        // Add rules so that Full Sequencer bootstrap messages are not sent to PORT_0.
+        TestRule dropFullBootstrapRule = new TestRule()
```

Comment:
To be on the safe side do you think we should add these rules before even clearing the server rules on PORT_0? I know there shouldn't be a change in the sequencer but if it were , we are sure it will not be taking a full bootstrap message.

Reply:
It doesn't matter where we add it as long as we add it before we partition PORT_1 and mark it unresponsive. Once PORT_1 is partitioned out then PORT_0 should not receive full bootstrap messages for the test case to work. Does this explanation help?

Reply:
yes that's why I said there shouldn't be a sequencer change even when PORT_0 rules are cleared, but was wondering if coming back could in any way trigger a change of sequencer even when PORT_! is still up. If there is no possible way for this to happen, I agree the order does not matter.

Reply:
Right. I'm moving this to the top just to be safe.